torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -2054,9 +2054,9 @@ void decomposeLinearOps(Block* block) {
     // TODO: The assert is not necessary when we can handle matmul, right now we
     // are splitting the linear between matmul & bias_add. Our fuser can only
     // take the second half and we would need the size information.
-    TORCH_INTERNAL_ASSERT(
-        mat0_size.has_value() && mat1_size.has_value(),
-        "concrete shape for linear input & weight are required");
+    if (!mat0_size.has_value() || !mat1_size.has_value()) {
```

Comment:
Do you think it's ok that we throw a one time warning here? Or maybe even only throw warning when we are in debug.
The assert was put here so that we would be well aware of missing profiling information.